core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java

```diff
@@ -380,15 +380,22 @@ private void validateStackAuthentication(StackV4Request source) {
     }
 
     private boolean shouldCreateInternalKnoxLoadBalancer(StackType type, EnvironmentNetworkResponse network) {
-        // TODO expand this to data hubs
+        return shouldCreateInternalKnoxLoadBalancerForDatalake(type, network) ||
+            shouldCreateInternalKnoxLoadBalancerForDatahub(type, network);
+    }
+
+    private boolean shouldCreateInternalKnoxLoadBalancerForDatalake(StackType type, EnvironmentNetworkResponse network) {
         return StackType.DATALAKE.equals(type) &&
             (entitlementService.datalakeLoadBalancerEnabled(ThreadBasedUserCrnProvider.getAccountId())
             || isEndpointGatewayEnabled(network));
     }
 
+    private boolean shouldCreateInternalKnoxLoadBalancerForDatahub(StackType type, EnvironmentNetworkResponse network) {
+        return  StackType.WORKLOAD.equals(type) && isEndpointGatewayEnabled(network);
```

Comment:
Shouldn't this also consider an entitlement for {{WORKLOAD}} or that entitlement has already been eliminated?

Reply:
Enis and I had a talk about that. The reason there's a DATALAKE specific entitlement is because the load balancer is used for both the public endpoint gateway logic, and for data lake HA. So I wanted a way to enable or disable one of those features at a time. But for data hubs, the load balancer is only used in the public endpoint gateway feature. So we decided to just use that entitlement to gate the data hub load balancer.

Reply:
Thanks, I tracked it down from the code and approved the PR, only comment is to squash the changes into one commit.